src/components/ViewSnippetHeader.tsx

```diff
@@ -16,6 +16,7 @@ export default function ViewSnippetHeader() {
   const axios = useAxios()
   const qc = useQueryClient()
   const session = useGlobalStore((s) => s.session)
+  const [isStarred, setIsStarred] = useState(snippet?.is_starred || false)
```

Comment:
don't use `useState` for the is_starred state, instead use an optimistic update or just wait for the invalidation

https://tanstack.com/query/v4/docs/framework/react/guides/optimistic-updates#updating-a-list-of-todos-when-adding-a-new-todo